### ISSUE TITLE
Add an option to set rel=nofollow to the links

### ISSUE THREAD

**Issue by kkomelin** [backdrop-contrib/socialfield#13]
I would like to have an option to add rel=nofollow attribute to the links so that search engines cannot follow the links.

I am going to send a PR with a change similar to https://www.drupal.org/node/2751147#comment-11306373 Just need to perform more tests.

By the way, I saw this module on BackdropCMS site first time in my life, and now I am using it on my sites. Thanks for that and for porting this module to Backdrop.

**Comment by CybersecDan** [comment]
Closing this per #14.

**Comment by jenlampton** [comment]
Reopening because I have a question :)  Sorry to be late to the conversation here... but why does this need to be an option? Shouldn't all these links always be rel=nofollow? Can you please state a case for when you want them to have nofollow and when you don't?

I'm only asking because I'm worried about user experience. Will people know when items should be nofollowed and when they shouldn't? If we can make that decision for them (and be right at least 80% of the time) maybe we don't need a setting for it a all?

Food for thought!

**Comment by klonos** [comment]
When we have such discussions about 80/20 use cases, I can't help but to think that by setting the default to be what the 80% expects but at the same time having a way for the rest of the 20% to easily change the default is the best way to go about things. Just my 2c after taking a "thought bite" 😄

**Comment by jenlampton** [comment]
comment from @kkomelin in https://github.com/backdrop-contrib/socialfield/pull/14#issuecomment-263485113: 

> Hi @jenlampton,

> I implemented the rel=nofollow as an option because I always try to take into account and respect current users's interests. I can't just turn on/off a feature that can potentially negatively affect their SEO rank (even indirectly). That's why rel=nofollow is turned off by default, and it doesn't change anything for existing users.

> However, less number of choices leads to better usability and I believe users appreciate it in Backdrop. My only consideration is that any improvement should not harm existing users.

> On top of that, as a Backdrop founder, you might have own vision, so feel free to decide. I will support your decision.

**Comment by jenlampton** [comment]
> I always try to take into account and respect current users's interests.

Ah, was this a setting on the Drupal module that didn't make it into the Backdrop version? Are you porting a site from Drupal to Backdrop? Is your use-case one that couldn't have added the rel=nofollow links in the theme? I'm still curious to know more about why a setting was necessary for your use-case.

Drupal tends to suffer from developers who want to please everyone, and instead of making decisions about what options should exist (and what shouldn't), they often build interfaces for every possible permutation - which leads to a bad user experience in the long run.

In Backdrop core we try to engineer the User-Interface for the 80% use-cases only, and if there is another way to solve a problem faced by only 20% of users (like a theme override) then there's no reason to clutter the UI with it. I've been trying to apply the same reasoning to contrib wherever possible.

**Comment by jenlampton** [comment]
Also comment from @danielrose28 👍 

> Wish I had more of a case for it either way. I think it's useful, but honestly can't think of a reason for it not to be. I know what it's for but haven't done enough with it to give pros/cons, etc. Feel free to school me on this guys. :)

**Comment by jenlampton** [comment]
And my reply: 

> Some light reading:
https://moz.com/community/q/is-it-better-nofollow-or-follow-links-to-external-social-pages
https://beymour.com/when-you-should-and-shouldnt-use-a-nofollow
http://sixrevisions.com/html/rel-nofollow/

**Comment by jenlampton** [comment]
@kkomelin I just re-read your comment, and I believe I better understand now. When you said `existing users` you were talking about Backdrop users - those already using the module. You didn't want to change their sites by adding `rel=nofollow` to all links. Gotcha!

**Comment by jenlampton** [comment]
Well, from some quick googling it looks like it's generally a bad idea to add `rel=nofollow` links to social media, and that makes me think it definitely shouldn't be a setting. 

That said, from looking over the code it appears that there isn't any **other** way to make this change on a site that *does* need it. There's no way to theme override with the way things are written - and that is our real problem. Not having a decent theme override is going to present all sorts of other kinds of issues too, so we should fix that, and maybe it solves this problem too. (see https://github.com/backdrop-contrib/socialfield/issues/16)

**Comment by kkomelin** [comment]
Hi @jenlampton,

I wrote:

> I always try to take into account and respect current users's interests.

You answered:

> Ah, was this a setting on the Drupal module that didn't make it into the Backdrop version? Are you porting a site from Drupal to Backdrop? Is your use-case one that couldn't have added the rel=nofollow links in the theme? I'm still curious to know more about why a setting was necessary for your use-case.

I was talking in general. That's why used the word "always".
I made a patch for the Drupal module and decided to share it with Backdrop community as well https://www.drupal.org/node/2751147

You wrote:

> Drupal tends to suffer from developers who want to please everyone, and instead of making decisions about what options should exist (and what shouldn't), they often build interfaces for every possible permutation - which leads to a bad user experience in the long run.

I wouldn't like to compare Backdrop with Drupal but I could understand why you would do so. And I mentioned that a quantity of choices can affect usability:

> However, less number of choices leads to better usability and I believe users appreciate it in Backdrop. My only consideration is that any improvement should not harm existing users.

You then wrote:

> @kkomelin I just re-read your comment, and I believe I better understand now. When you said existing users you were talking about Backdrop users - those already using the module. You didn't want to change their sites by adding rel=nofollow to all links. Gotcha!

Sorry if my comment was not clear. Now you understand it correctly.

> Well, from some quick googling it looks like it's generally a bad idea to add rel=nofollow links to social media, and that makes me think it definitely shouldn't be a setting.

Here is a use case when rel=nofollow necessary, at least for me. You're an event site owner and you'd like to make search engines "nofollow" social links of your users (a big number of users). 

To sum up, the rel=nofollow implemented as an option because:
1) Current users of the Backdrop module can continue using it without any changes. Nothing will change for them until they want it.
2) The rel=nofollow option is necessary in some cases, not in all cases.

However, as I have already mentioned:
> ... as a Backdrop founder, you might have own vision, so feel free to decide.

Unfortunately, I can't afford to spend more time on the matter, so:
> I will support your decision.

**Comment by CybersecDan** [comment]
After doing some reading myself, I can understand not wanting to add "nofollow", so that makes sense. I also understand the need for wanting to add it. It really does depend. So it would make sense to at least give the option to the user to add it if they feel like they'd need it. But as @kkomelin stated as well, I'm on board with your decision @jenlampton. Both of you have more experience than I do and know better, so I'll support either decision. 

I know, sorry for just agreeing with everyone. :)

**Comment by jenlampton** [comment]
If all use-cases are valid then we need to make sure they are all possible. They weren't before, but they should be now using the theme-patch.  The next question is, is this an 80% use-case? I honestly don't know the answer to that!  

I think we should wait and see if this issue gets any other attention, and if it does, and if it starts to sound like the answer to that question is YES, then we should add it as a setting :)

**Comment by CybersecDan** [comment]
Should we just close this issue for now? If this topic gets any attention in the future we can always reopen it. Just asking since there hasn't been any attention since last November. I'm fine with keeping it open. Just thought I'd check in.

Also wanted to reference the PR that was merged today, #17, since this issue is still open, people know it's being addressed. :)

**Comment by jenlampton** [comment]
let's leave it for now :)

**Comment by jenlampton** [comment]
Since there hasn't been any movement on this since 2017, closing. But we can re-open if people start to feel a need for this feature :)
